addons/mail/models/template_reset_mixin.py

```diff
@@ -90,11 +90,15 @@ def reset_template(self):
         user lang (all langs are not supported due to costly file operation). """
         expr = "//*[local-name() = $tag and (@id = $xml_id or @id = $external_id)]"
         templates_with_missing_source = []
+        lang_false = {code: False for code, _ in self.env['res.lang'].get_installed() if code != 'en_US'}
         for template in self.filtered('template_fs'):
             external_id = template.get_external_id().get(template.id)
             module, xml_id = external_id.split('.')
             fullpath = get_resource_path(*template.template_fs.split('/'))
             if fullpath:
+                for field_name, field in template._fields.items():
+                    if field.translate is True:
+                        template.update_field_translations(field_name, lang_false)
```

Comment:
So if I understand well, you reset all translated fields, and they are then populated again with the right value if present ? As it does a cr.execute for each field (if I understand the code of '_update_field_translations' well), isn't it going to make unnecessary db update when we have a new value that is going to populate the translation afterwards anyway ?

Reply:
Strictly speaking, you are right
But we usually don't know if a translation is valid/translated in the po file.
If you `have a new value that is going to populate the translation afterwards anyway` I think you don't need to `reset_template`, just override translation directly

And since we cannot void a translation using the cache and `flush` (ORM limitation), using `update_field_translations` is the only choice currently to promise the `reset` logic
Since fetching several translations by parsing the whole po file is already very slow enough, I thought those extra SQL `UPDATE` shouldn't be a huge problem